src/backend/gl/src/native.rs

```diff
@@ -38,6 +43,79 @@ impl Fence {
     }
 }
 
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+pub(crate) enum BindingTypes {
+    UniformBuffers,
+    StorageBuffers,
+    StorageImages,
+    SampledImages,
+    SeparateImages,
+    SeparateSamplers,
+    SubpassInputs,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct DescRemapData {
+    bindings: HashMap<BindingTypes, HashMap<SetID, HashMap<BindingID, BindingID>>>,
+    names: HashMap<String, (BindingTypes, SetID, BindingID)>,
+    next_binding: BindingID,
+}
+
+impl DescRemapData {
+    pub(crate) fn new() -> Self {
+        DescRemapData {
+            bindings: HashMap::new(),
+            names: HashMap::new(),
+            next_binding: 0,
+        }
+    }
+
+    pub(crate) fn get_binding_insert(&mut self, btype: BindingTypes, set: SetID, binding: BindingID) -> BindingID {
+        if !self.bindings.contains_key(&btype) {
```

Comment:
let's use the [Entry API](https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html#method.entry)